lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -133,6 +150,90 @@ class QcFlagService {
         }
     }
 
+    /**
+     * Get QC summary for given data/simulation pass
+     * @param {number} dataPassId data pass id - exclusive with simulationPassId
+     * @param {number} simulationPassId simulation pass id - exclusive with dataPassId
+     * @return {Promise<QcSummary[]>} summary
+     */
+    async getQcFlagsSummary({ dataPassId, simulationPassId }) {
+        if (!dataPassId && !simulationPassId) {
+            throw new BadParameterError('Cannot create QC flag without data pass or simulation pass');
```

Comment:
Please update the error messages as per the use case

Reply:
updated